### PR TITLE
add start-group and end-group around php-config libs

### PR DESCRIPTION
**Pull request by DubbleClick**
don't be dependent on php not messing up when compiling the embed sapi

### CODE REVIEW

**Review by dunglas — APPROVED** [review]
Nice one!

Shouldn't we also update the Mac build to do the same thing?

**Comment by DubbleClick** [comment]
Oh good catch, I didn't touch the github actions.

But I have a question now, the dev-alpine.Dockerfile only builds php in zts mode with no extensions - can you use dynamically loaded extensions from package managers then (apt-get, rpm, etc)?

File: .github/workflows/static.yml [hunk 1]
```diff
@@ -111,7 +111,7 @@ jobs:
         run: |
            if [ "$GITHUB_REF_TYPE" == "tag" ]; then export FRANKENPHP_VERSION=${GITHUB_REF_NAME:1}; else export FRANKENPHP_VERSION=$GITHUB_SHA; fi
            echo "CGO_CFLAGS=-DFRANKENPHP_VERSION=$FRANKENPHP_VERSION $(./buildroot/bin/php-config --includes | sed s#-I/#-I$PWD/buildroot/#g)" >> "$GITHUB_ENV"
-           echo "CGO_LDFLAGS=-framework CoreFoundation -framework SystemConfiguration $(./buildroot/bin/php-config --ldflags) $(./buildroot/bin/php-config --libs)" >> "$GITHUB_ENV"
+           echo "CGO_LDFLAGS=-framework CoreFoundation -framework SystemConfiguration $(./buildroot/bin/php-config --ldflags) -Wl,--start-group $(./buildroot/bin/php-config --libs) -Wl,--end-group" >> "$GITHUB_ENV"
```
Comment (dunglas) [inline .github/workflows/static.yml:114]: There is also this line to update (we have to use CircleCI for Apple Silicon builds): https://github.com/dunglas/frankenphp/blob/main/.circleci/config.yml#L42C108-L42C108

**Comment by dunglas** [comment]
The `dev-alpine.Dockerfile` file is only intended for contributors/maintainers of FrankenPHP. It is totally internal. It just helps set up a dev environment with GDB and the like. working properly.

For general use of non-static builds, it's better to use Docker or [Setup PHP](https://github.com/dunglas/frankenphp/blob/main/.github/workflows/tests.yml) builds (which can also installed using Homebrew etc, and provides a lot of extensions already compiled for ZTS). Documenting how to use Setup PHP with FrankenPHP is on my todolist.

File: .github/workflows/static.yml (hunk 1 above)
Comment (DubbleClick) [reply .github/workflows/static.yml:114]: I'm not familiar with the linker on Apple Silicon. Do you know if it's necessary? New linkers typically already search libs in both directions.

Comment (dunglas) [reply .github/workflows/static.yml:114]: I don't know 😅 It's definitely a recent version of clang/LLVM.
However, what's sure is that it's the same version of clang/LLVM on GitHub Actions (even if running on Intel) and CircleCI. So if it's not necessary, it's not necessary to update the docs and the GHA Mac build either.

Comment (DubbleClick) [reply .github/workflows/static.yml:114]: I'll assume it's still using gnu-ld, can't find reliable information online and don't have a mac to test it on.

Comment (dunglas) [reply .github/workflows/static.yml:114]: It's using `ldd`, the LLVM linker: https://lld.llvm.org/

Comment (DubbleClick) [reply .github/workflows/static.yml:114]: > I don't know 😅 It's definitely a recent version of clang/LLVM. However, what's sure is that it's the same version of clang/LLVM on GitHub Actions (even if running on Intel) and CircleCI. So if it's not necessary, it's not necessary to update the docs and the GHA Mac build either.

gcc/gnu-ld never changed the default functionality, even on recent versions, that's why they still have `-( -)` or `--start-group --end-group`. gold, lld and vc switched to it by default (or released with it).

**Comment by DubbleClick** [comment]
>>> LLD remembers the symbol table of archives that it has previously seen, so if there is a reference from an input file to the right of an archive, LLD will still search that archive for resolving any undefined references. This means that an archive only needs to be included once on the command line and the --start-group and --end-group options are redundant.

https://lld.llvm.org/ELF/warn_backrefs.html

Not needed on MacOS then.

**Comment by dunglas** [comment]
Thank you very much @DubbleClick!

**Pull request merged**